src/coreclr/System.Private.CoreLib/src/System/MulticastDelegate.CoreCLR.cs

```diff
@@ -167,7 +167,8 @@ private static bool TrySetSlot(object?[] a, int index, object o)
         private MulticastDelegate NewMulticastDelegate(object[] invocationList, int invocationCount, bool thisIsMultiCastAlready)
         {
             // First, allocate a new multicast delegate just like this one, i.e. same type as the this object
-            MulticastDelegate result = InternalAllocLike(this);
+            // No need to check for type being initialized since we are making a copy of an existing instance.
+            MulticastDelegate result = (MulticastDelegate)RuntimeTypeHandle.InternalAlloc((RuntimeType)GetType(), checkTypeFullyInitialized: false);
```

Comment:
Is the `checkTypeFullyInitialized: false` micro-optimization still worth it now that it passed through multiple layers as a flag? I think it should be ok to get rid of it. 

If necessary, I would tweak `MethodTable::Allocate` to make to streamline the steady state path to avoid regressions.